### PR TITLE
Support timeout event. Increase timeout value

### LINKED ISSUE #9: Update step timeout value on any occurrence of timeout event
**Describe the bug**
 None of the steps process timeout event. 

```golang

			// in case of timeout we increase the timeout and that's it
			p.IncreaseTimeout(r.Round)
```

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #9

### CODE REVIEW

File: src/msg_handler.rs [hunk 1]
```diff
@@ -10,17 +10,24 @@ use crate::user::committee::Committee;
 use hex::ToHex;
 use std::fmt::Debug;
 
+pub struct HandleMsgOutput {
+    pub result: Message,
+
+    // if true, it instructs the event loop to bubble up the message and terminate.
+    pub is_final_msg: bool,
+}
+
```
Comment (herr-seppia) [inline src/msg_handler.rs:19]: Why not use an Enum?

A sort of 
```rust
pub enum HandleMsgOutput {
    Result(Message),
    FinalResult(Message)
}
```

Looking at your code seems to fit enough and it's more ideomatic

File: src/phase.rs [hunk 2]
```diff
@@ -76,4 +76,8 @@ impl Phase {
     fn name(&self) -> &'static str {
         call_phase!(self, name())
     }
+
+    fn get_timeout(&self) -> u64 {
+        call_phase!(self, get_timeout())
```
[the remark this answers is not in the document]
Comment (goshawk-3) [reply src/phase.rs:81]: Any suggestions?

File: src/msg_handler.rs (hunk 1 above)
Comment (goshawk-3) [reply src/msg_handler.rs:19]: That would be indeed more idiomatic. I'll address it next PR.

Comment (goshawk-3) [reply src/msg_handler.rs]: Actually, I liked that a lot and  I addressed it with this PR. Thanks!

**Pull request merged**